src/components/LoadingIndicator/LoadingIndicator.jsx

```diff
@@ -1,12 +1,28 @@
 import PropTypes from 'prop-types';
 import React from 'react';
+import dispatchAnalyticsEvent from '../../helpers/analytics';
 
 export default class LoadingIndicator extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { loadingStartTime: Date.now() };
```

Comment:
Can you add a comment here stating that this state variable is more like a `const`, and that it is only used to get a time difference between initial mount and unmount?